django_mesh/tests/test_models.py

```diff
@@ -35,32 +48,92 @@ def test_has_custom_summary(self):
         self.assertNotEqual(self.p1.summary, self.p1.teaser)
         self.assertEqual(self.p1.summary, self.p1.custom_summary)
 
-    def test_render_stays_the_same_when_no_links(self):
-        self.assertNotEqual(self.p1.text, '')
-        self.assertEqual(self.p1.rendered_text, '')
+    def test_markup(self):
+        self.c1.save()
+
+        self.p1.channel = self.c1
+        self.p7.channel = self.c1
+        self.p8.channel = self.c1
 
+        self.p1.text = 'hello'
+        self.p7.text = 'hello'
+        self.p8.text = 'hello'
+
+        self.p1.save() # plaintext
+        self.p7.save() #markdown
+        self.p8.save() #textile
+
+        self.assertEqual(self.p1.text, self.p1.rendered_text)
+        self.assertEqual(markdown.markdown(self.p7.text), self.p7.rendered_text)
+        self.assertEqual(textile.textile(self.p8.text), self.p8.rendered_text)
+
+    def test_plain_text(self):
         self.c1.save()
         self.p1.channel = self.c1
+        self.p1.text = '<a href=www.google.com> google link </a>'
+        self.p1.slug = 'testing_html_escape'
         self.p1.save()
+        response = self.client.get(reverse('mesh_post_view', kwargs={'slug': self.p1.slug}))
 
-        self.assertEqual(self.p1.rendered_text, self.p1.text)
+        self.assertEqual(response.status_code, 200)
 
-    def test_render_adds_anchors(self):
-        self.p1.text = """foo
-            http://somelink
-            http://anotherlink
-        """
+        self.assertNotIn('<', self.p1.rendered_text)
+        self.assertNotIn('>', self.p1.rendered_text)
+        self.assertContains(response, '<')
+        self.assertContains(response, '>')
 
+    def test_rendered_text_stays_the_same_when_no_links(self):
+        self.assertNotEqual(self.p1.text, '')
         self.assertEqual(self.p1.rendered_text, '')
 
         self.c1.save()
         self.p1.channel = self.c1
         self.p1.save()
 
-        self.assertEqual(self.p1.rendered_text, """foo
-            <a href="http://somelink">http://somelink</a>
-            <a href="http://anotherlink">http://anotherlink</a>
-        """)
+        self.assertEqual(self.p1.rendered_text, self.p1.text)
+
+    def test_automatic_anchors_to_links_on_own_line_if_markdown_chosen(self):
+
+        self.p7.text = "foo\n   \n http://somelink\n" 
+
+        self.assertEqual(self.p7.rendered_text, '')
+
+        self.c1.save()
+        self.p7.channel = self.c1
+        self.p7.save()
+
+        anchor = '<a href="http://somelink"'
+
+        self.assertIn(anchor, self.p7.rendered_text,)
```

Comment:
Let's go ahead and test the full rendered_text. This should make the assertion a little less error prone:
https://docs.djangoproject.com/en/1.7/topics/testing/tools/#django.test.SimpleTestCase.assertHTMLEqual